src/cli/index.js

```diff
@@ -64,27 +66,27 @@ function runCruise(pFileDirArray, pOptions) {
 
     io.write(pOptions.outputTo, lDependencyList.modules);
 
-    /* istanbul ignore if */
     if (lDependencyList.summary.error > 0) {
-        process.exit(lDependencyList.summary.error);
+        lExitCode = lDependencyList.summary.error;
     }
+    return lExitCode;
 }
 
 module.exports = (pFileDirArray, pOptions) => {
     pOptions = pOptions || {};
+    let lExitCode = 0;
 
     try {
         if (pOptions.info === true) {
             process.stdout.write(formatMetaInfo());
         } else if (pOptions.init === true){
             createRulesFile(pOptions);
         } else {
-            runCruise(pFileDirArray, pOptions);
+            lExitCode = runCruise(pFileDirArray, pOptions);
         }
     } catch (e) {
         process.stderr.write(`\n  ERROR: ${e.message}\n`);
+        lExitCode = -1;
```

Comment:
why `-1`? AFAICT (all common) shells use one byte unsigned integers as return codes. That means -1 is simply converted to 255.

Reply:
No particular reason, just an old habit. It idd results in 255 on un*x/ posix platforms.

I'll see if there's something more appropriate - I'll probably settle for `1`. An unrepresentative sample from some of the tools in ts/js land: tsc, tslint, acorn use `1`, eslint uses `2`. (Suggestions welcome).